### PR TITLE
configs: Rawhide to accept GPG key from future Fedora Rawhide+1

### PR DESCRIPTION
**Pull request by praiskup**
Fixes: #1338

### CODE REVIEW

File: mock-core-configs/etc/mock/templates/fedora-rawhide.tpl [hunk 1]
```diff
@@ -42,7 +42,7 @@ user_agent={{ user_agent }}
 
 {%- macro rawhide_gpg_keys() -%}
 file:///usr/share/distribution-gpg-keys/fedora/RPM-GPG-KEY-fedora-$releasever-primary
-{%- for version in [releasever|int, releasever|int - 1]
+{%- for version in [releasever|int, releasever|int - 1, releasever|int + 1]
```
Comment (keszybz) [inline mock-core-configs/etc/mock/templates/fedora-rawhide.tpl:45]: Thanks! My only qualm is what happens if the key doesn't exist yet? We don't want any kind of warning or error in that case. Hmm, I think we get an error if they key doesn't exist:
```
cannot open file: (2) - No such file or directory [/usr/share/distribution-gpg-keys/fedora/RPM-GPG-KEY-fedora-46-primary]
```

I think this should be wrapped in a check whether the key actually exists.

Comment (praiskup) [reply mock-core-configs/etc/mock/templates/fedora-rawhide.tpl:45]: Good point, I installed this fatal error into the branching script:

```
File /usr/share/distribution-gpg-keys/fedora/RPM-GPG-KEY-fedora-43-primary not found!

The new Fedora Rawhide (42) requires a GPG key for Fedora 43,
which must be generated by Fedora Infrastructure team and shipped in the
distribution-gpg-keys.rpm package.  This either has not happened yet, the
package is outdated, or it is not even installed on this box.
```

**Comment by praiskup** [comment]
This time, F42 key was generated [on 2024-02-12](https://src.fedoraproject.org/rpms/fedora-repos/c/d10a092adbcf0e0b2d7e4f05311255c805c878b5).
The plan was to have it about [~14 days before signing rawhide with f41 key](https://fedorapeople.org/groups/schedule/f-40/f-40-all-tasks.html), which IMO happened (the mass rebuild took a bit more time). Not sure.
Mock-core-configs with branched F40 [2024-02-14](https://github.com/rpm-software-management/mock/releases/tag/mock-core-configs-40.1-1). Which was admittedly too late, it would be nice to do that 14 days earlier next time.

**Comment by xsuchy** [comment]
I have to say that I am not happy for this PR. IMO this will bring a new issue when we submit new config earlier than before branching (missing key). 
I would rather resolve it by dynamically using number for rawhide. Likely using python-fedora-distro-aliases https://github.com/rpm-software-management/fedora-distro-aliases  In such case we would not need to touch rawhide config after branching at all!

**Comment by praiskup** [comment]
"fedora-repos update with the new keys is available in updates-testing" is though something different, I think that we should have a new entry in the F41 schedule like

    Generate Fedora N+2 key and ship in fedora-repos

Somewhere in the Fedora 41 schedule.  To have a guidance that assures everything works.

**Comment by praiskup** [comment]
> I have to say that I am not happy for this PR. IMO this will bring a new issue
> when we submit new config earlier than before branching (missing key).

If we can get a new step into the Fedora release schedule, then this problem
would be non-existing.

> I would rather resolve it by dynamically using number for rawhide. Likely
> using python-fedora-distro-aliases
> https://github.com/rpm-software-management/fedora-distro-aliases In such case
> we would not need to touch rawhide config after branching at all!

This is being tracked here #1328 - though :-/ it's going to be a big change in
the way mock and core-configs behave.

**Comment by xsuchy** [comment]
I tried this with DNF4 and DNF5 and it prints no error when the file does not exists and it is silently ignored. Not sure if this is generally good, but for our case it is good.
I changed my mind and I am +1

**Comment by xsuchy** [comment]
I manually rebased this and merged as 66ca1e4b

**Pull request closed without merging**